notebooks/roc_plot_generator.py

```diff
@@ -299,214 +376,301 @@ def sp_run_filter(run): # name):
 exp = TLACDCExperiment(
     model=things.tl_model,
     threshold=100_000,
-    early_exit=True,
+    early_exit=SKIP_ACDC or ONLY_SAVE_CANONICAL,
     using_wandb=False,
-    zero_ablation=False,
-    ds=things.validation_data,
-    ref_ds=things.validation_patch_data,
+    zero_ablation=bool(ZERO_ABLATION),
+    ds=things.test_data,
+    ref_ds=things.test_patch_data,
     metric=things.validation_metric,
     second_metric=None,
     verbose=True,
-    use_pos_embed=use_pos_embed,
+    use_pos_embed=USE_POS_EMBED,
+    first_cache_cpu=False,
+    second_cache_cpu=False,
 )
+if not SKIP_ACDC and not ONLY_SAVE_CANONICAL:
+    exp.setup_second_cache()
+
+max_subgraph_size = exp.corr.count_no_edges()
+
 #%% [markdown]
 # Load the *canonical* circuit
 
-d = {(d[0], d[1].hashable_tuple, d[2], d[3].hashable_tuple): False for d in exp.corr.all_edges()}
-d_trues = get_true_edges()
-for k in d_trues:
-    d[k] = True
-exp.load_subgraph(d)
-canonical_circuit_subgraph = deepcopy(exp.corr)
-canonical_circuit_subgraph_size = canonical_circuit_subgraph.count_no_edges()
-
+if TASK != "induction":
+    d = {(d[0], d[1].hashable_tuple, d[2], d[3].hashable_tuple): False for d in exp.corr.all_edges()}
+    d_trues = get_true_edges()
+    if ONLY_SAVE_CANONICAL and TASK == "ioi":
+        # Remove non-adjacent layer connections
+        def layer(name):
+            return int(name.split(".")[1])
+
+        for t in d_trues.keys():
+            if abs(layer(t[0]) - layer(t[2])) > 1:
+                del d_trues[t]
+
+    for k in d_trues:
+        d[k] = True
+    exp.load_subgraph(d)
+    canonical_circuit_subgraph = deepcopy(exp.corr)
```

Comment:
```suggestion
    canonical_circuit_subgraph = deepcopy(exp.corr)
    for t, e in exp.corr.all_edges():
        exp.corr.edges[t[0]][t[1]][t[2]][t[3]].present = True
```

This is crucial, as it fixes a bug where some runs looked like they were perfect recoveries, but instead they had just crashed

Reply:
Good catch, I completely overlooked that!

Reply:
The crashed runs should only be retrieved for ACDC, I hope they were ACDC runs.